configuration.k

```diff
@@ -0,0 +1,220 @@
+                    // Gas: not sure how, but I think we need it. 
+                    <gas>         .K          </gas>
+                    <memoryUsed>  .K          </memoryUsed>
+                    <previousGas> .K          </previousGas>
+
+                    // Those are present in both KEVM and IELE. 
+                    // Not sure we need them, let's keep them around until figure it out. 
+                    <static>    .K     </static>
+                    <callDepth>   0    </callDepth>
+                    <localCalls> .List </localCalls>
+                </callState>
+
+                // A_* (execution substate)
+                // (same as KEVM/IELE)
+                <substate>
+                    <selfDestruct> .Set  </selfDestruct>            // A_s
+                    <log>          .List </log>                     // A_l
+                    <refund>       0     </refund>                  // A_r
+                </substate>
+
+                // Immutable during a single transaction
+                // -------------------------------------
+
+                <gasPrice> 0 </gasPrice>                          // I_p
+                <origin>   0 </origin>                            // I_o
+
+                // I_H* (block information)
+                // Needed or not? It is part of KEEI, but not sure. Should discuss. 
+                <block>
+                    <hashes>           .List      </hashes>
+                    <coinbase>         0          </coinbase>         // H_c
+                    // <logsBloom>        .WordStack </logsBloom>        // H_b
+                    <difficulty>       0          </difficulty>       // H_d
+                    <number>           0          </number>           // H_i
+                    <gasLimit>         0          </gasLimit>         // H_l
+                    <timestamp>        0          </timestamp>        // H_s
+                </block>
+            </execEngine>
+            
+            // This is the part of the "Solidity VM" where we keep the available contracts. 
+            // Those are either obtained from source file (creation) or loaded from accounts (function call).
+
+            // When we parse Solidity source file we 
+            //   - desugar/preprocess modifiers, inheritance (?) etc. 
+            //   - create one <contract> structure for each contract declared in the sources
+            // (let's not worry about those for now, and just stick to simple functions)
+
+            // From here, <contract>s can be either deployed or executed. 
+            // When a contract is deployed we not only copy its <contract> cell into the <account>, 
+            // but we also copy the other contracts that were present in the same source file, 
+            // so that the deployed contract have access to them and can create and deploy them if needed. 
+            // (see the <account> cell below for more notes about this)
+
+            <contracts>
```

Comment:
it seems strange to me that this isn't part of the network cell. shouldn't all this just live inside the `<code>` part of an account?

Reply:
I agree with @dwightguth here, the `<Contract>` subcell can be directly inside the `<code>` cell, then we'll match the other blockchain definitions better but still get the benefits of having a more structured `<Contract>` representation of the code.

Reply:
I think it makes sense to put `<contracts>` outside `<network>` because not all contracts are deployed.
e.g.
https://github.com/gnosis/safe-contracts/blob/6db7ce0f600d391bc6f34b144d7845bf5f227d81/contracts/common/Enum.sol
https://github.com/gnosis/safe-contracts/blob/c942db2895fa0b1003531cabb26f8dc37c999ec1/contracts/base/Executor.sol#L12
The contract `Enum` is not deployed but its code is used in another contract `Executor`.

Reply:
The reason I suggested this approach is that I thought we'd need some temporary place to store those (processed) contract definitions before even deploying them (or using them for whatever other purpose). Let's say, for example, we load a source file; before deploying (or performing any other operation), the code has to be parsed, pre-processed (e.g. handle modifiers) and "packaged" into a data structure. Only after that, there will be a command saying "OK now deploy contract X (not the actual source, but the data structure) in address Y". So I think we need some "buffer" like that to store contracts before using them, so the idea is to only copy a contract in `<network>` if and when needed.

Reply:
Yes, exactly as @tomtomjhj  said. Not all contracts will be deployed. The network only includes all the contract instances that are deployed. So, I guess we keep `contracts` as where it is?

Reply:
Yes, leave as is for now.